browser/src/client_win.cc

```diff
@@ -61,21 +61,26 @@ int ClientWin::Send(const ContentAnalysisRequest& request,
   }
 
   Handshake handshake;
-  handshake.set_content_analysis_requested(true);
+  handshake.set_content_analysis_requested(request.has_content_data());
   
   bool success = false;
 
   if (WriteMessageToPipe(handle, handshake.SerializeAsString())) {
-    if (WriteMessageToPipe(handle, request.SerializeAsString())) {
-      Acknowledgement acknowledgement;
-      std::vector<char> buffer = ReadNextMessageFromPipe(handle);
-      if (response->ParseFromArray(buffer.data(), buffer.size())) {
-        acknowledgement.set_verdict_received(response->results_size() > 0);
-        success = true;
-      }
-      if (!WriteMessageToPipe(handle, acknowledgement.SerializeAsString())) {
-        success = false;
+    // If the message has no content data, only a handshake is sent
```

Comment:
For better readability, you may want to avoid using nested if. See https://g3doc.corp.google.com/company/readability/home/acquiring-readability/c-readability-advice.md?cl=head#avoid-unnecessary-nesting-by-using-early-returns